src/plugins/data/common/field_formats/field_formats_registry.ts

```diff
@@ -42,7 +41,7 @@ export class FieldFormatsRegistry {
   protected metaParamsOptions: Record<string, any> = {};
   protected getConfig?: FieldFormatsGetConfigFn;
   // overriden on the public contract
-  public deserialize: (mapping: SerializedFieldFormat) => IFieldFormat = () => {
+  public deserialize: FormatFactory = () => {
```

Comment:
The public contract uses `FormatFactory`, so this inconsistency caused some type issues when moving esaggs to the server.